op-node/node/node.go

```
@@ -124,6 +124,30 @@ func (n *OpNode) initL1(ctx context.Context, cfg *Config) error {
 		return fmt.Errorf("failed to create L1 source: %w", err)
 	}
 
+	// Validate the L1 Client Chain ID
+	id, err := n.l1Source.L1ChainID(ctx)
+	if err != nil {
+		n.log.Warn("failed to verify L1 RPC chain id", "err", err)
+	} else {
+		if cfg.Rollup.L1ChainID != id {
+			n.log.Warn("incorrect L1 RPC chain id", "expected", cfg.Rollup.L1ChainID, "actual", id)
```

Perhaps this should return an error, we don't want syncing inconsistencies

---

I think that both of these checks should return errors if they fail

---

i also think if the rpc request fails, it should error